packages/commons/src/big-query/liquidity-indices-table/pull-data/pullClosestIndices.ts

```diff
@@ -0,0 +1,54 @@
+import { getBigQuery } from '../../client';
+import { getTableFullName } from '../../utils/getTableName';
+import { TableType } from '../../types';
+import { Address } from '../../../utils/convertLowercase';
+
+export type LiquidityIndexEntry = {
+    chainId: number;
+    rateOracle: Address,
+    liquidityIndex: number,
+    blockTimestamp: number
+}
+
+const mapToLiquidityIndices = (row: any) : LiquidityIndexEntry => row;
+
+export const pullClosestIndices = async (
+  chainId: number,
+  rateOracle: Address,
+  targetTimestamp: number
+): Promise<LiquidityIndexEntry[] | null> => {
+  const bigQuery = getBigQuery();
+  const liquidityIndexTableName = getTableFullName(TableType.liquidity_indices);
+
+  const sqlQuery = `
+    WITH timestampBefore as (
+        SELECT min(${targetTimestamp} - blockTimestamp) AS before
+        FROM \`${liquidityIndexTableName}\` 
+        WHERE blockTimestamp <= ${targetTimestamp} AND chainId=${chainId} AND oracleAddress="${rateOracle}"
+    ),
+
+    timestampAfter as (
+        SELECT min(blockTimestamp - ${targetTimestamp}) AS after
+        FROM \`${liquidityIndexTableName}\` 
+        WHERE blockTimestamp >= ${targetTimestamp} AND chainId=${chainId} AND oracleAddress="${rateOracle}"
+    )
+    
+    SELECT blockTimestamp, chainId, oracleAddress as rateOracle, liquidityIndex
+    FROM \`${liquidityIndexTableName}\`
+    WHERE chainId=${chainId} AND oracleAddress="${rateOracle}" AND blockTimestamp = ${targetTimestamp} - (SELECT * FROM timestampBefore)
```

Comment:
can't we just say `... AND ( blockTimestamp = ${targetTimestamp} - (SELECT * FROM timestampBefore) OR blockTimestamp = ${targetTimestamp} + (SELECT * FROM timestampAfter))` ? This will save us one loop on the SQL side.

Reply:
truee, thankss! changed [e0c8edf](https://github.com/Voltz-Protocol/v2-off-chain-monorepo/pull/34/commits/e0c8edf872ac0156572fc067bdb129d70fc36e1d)